### ISSUE TITLE
(spooler / awsiotmqttclient): handle CLIENT_ERROR on message publish

### ISSUE THREAD

**Issue by kris-sum** [aws-greengrass/aws-greengrass-nucleus#852]
**Describe the bug**
If you send an MQTT message that results in CLIENT_ERROR , the connection gets reset (from the AWS IoT platform end), then the client reconnects and re-tries sending the message. This will then cause a CLIENT_ERROR, which causes a connection reset, and then the whole process happens again.

**To Reproduce**
Send an MQTT message using IPC with more than 7 slashes in the topic.

**Expected behavior**
The message should be discarded from the spooler on error.

**Actual behavior**
The message stays (in the spooler?) and is retried, causing an infinite loop.

**Environment**
OS: Raspbian GNU/Linux 10 (buster)
JDK version: openjdk version "1.8.0_212"
Nucleus version: 2.0.4

**Additional context**
As the mqttclient continually retries sending the broken message, no other MQTT messages can be received as the connection to the platform gets reset very quickly - this includes shadow device subscriptions, so it could potentially cause the greengrass device to become unresponsive. 

Log: 

```
2021-02-09T17:15:00.060Z [WARN] (Thread-5) com.aws.greengrass.mqttclient.AwsIotMqttClient: Connection interrupted. {clientId=RaspiCore-1, error=The connection was closed unexpectedly.}
2021-02-09T17:15:00.264Z [INFO] (Thread-5) com.aws.greengrass.mqttclient.AwsIotMqttClient: Connection resumed. {clientId=RaspiCore-1, sessionPresent=true}
2021-02-09T17:15:00.265Z [INFO] (pool-2-thread-24) com.aws.greengrass.deployment.ShadowDeploymentListener: Published to get named shadow topic. {}
2021-02-09T17:15:00.265Z [DEBUG] (pool-2-thread-23) com.aws.greengrass.deployment.IotJobsHelper: Requesting the next deployment. {ThingName=RaspiCore-1}
2021-02-09T17:15:00.266Z [DEBUG] (pool-2-thread-24) com.aws.greengrass.deployment.DeploymentStatusKeeper: Updating status of persisted deployments to subscribers. {deploymentType=SHADOW, numberOfSubscribers=2}
2021-02-09T17:15:00.266Z [DEBUG] (pool-2-thread-23) com.aws.greengrass.deployment.DeploymentStatusKeeper: Updating status of persisted deployments to subscribers. {deploymentType=IOT_JOBS, numberOfSubscribers=2}
2021-02-09T17:15:00.288Z [WARN] (Thread-5) com.aws.greengrass.mqttclient.AwsIotMqttClient: Connection interrupted. {clientId=RaspiCore-1, error=The connection was closed unexpectedly.}
2021-02-09T17:15:01.486Z [INFO] (Thread-5) com.aws.greengrass.mqttclient.AwsIotMqttClient: Connection resumed. {clientId=RaspiCore-1, sessionPresent=true}
2021-02-09T17:15:01.488Z [INFO] (pool-2-thread-24) com.aws.greengrass.deployment.ShadowDeploymentListener: Published to get named shadow topic. {}
2021-02-09T17:15:01.488Z [DEBUG] (pool-2-thread-23) com.aws.greengrass.deployment.IotJobsHelper: Requesting the next deployment. {ThingName=RaspiCore-1}
2021-02-09T17:15:01.489Z [DEBUG] (pool-2-thread-24) com.aws.greengrass.deployment.DeploymentStatusKeeper: Updating status of persisted deployments to subscribers. {deploymentType=SHADOW, numberOfSubscribers=2}
2021-02-09T17:15:01.489Z [DEBUG] (pool-2-thread-23) com.aws.greengrass.deployment.DeploymentStatusKeeper: Updating status of persisted deployments to subscribers. {deploymentType=IOT_JOBS, numberOfSubscribers=2}
2021-02-09T17:15:01.508Z [WARN] (Thread-5) com.aws.greengrass.mqttclient.AwsIotMqttClient: Connection interrupted. {clientId=RaspiCore-1, error=The connection was closed unexpectedly.}
```

**Comment by MikeDombo** [comment]
Thank you Kris,
We are aware of this behavior as we'll be addressing it in our next release.

**Comment by MikeDombo** [comment]
This issue has been addressed by #862 and it will be included in an upcoming release.
